File: KadirliApp/API_Deneme/Models/Campaign.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API_Deneme.Models
{
    public class Campaign
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [MaxLength(500)]
        public string Title { get; set; } = string.Empty;
        
        [Required]
        [Column("business_name")]
        [MaxLength(200)]
        public string BusinessName { get; set; } = string.Empty;
        
        public string? Description { get; set; }
        
        [Column("discount_code")]
        [MaxLength(50)]
        public string? DiscountCode { get; set; }
        
        [Column("image_urls", TypeName = "TEXT")]
        public string? ImageUrls { get; set; } // JSON array olarak saklanacak
        
        [Column("created_at")]
        public DateTime CreatedAt { get; set; }
    }
}

File: KadirliApp/KadirliApp/Domain/Entities/Campaign.swift
import Foundation

struct Campaign: Identifiable, Codable {
    let id: UUID
    let title: String
    let businessName: String
    let description: String?
    let discountCode: String?
    let imageUrls: [String]?
    let createdAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id, title, description
        case businessName = "business_name"
        case discountCode = "discount_code"
        case imageUrls = "image_urls"
        case createdAt = "created_at"
    }
}
